### PR TITLE
catalog: Remove SerializedX items from catalog

### PR DESCRIPTION
**Pull request by jkosh44**
Previously, the catalog had structs of the form 'SerializedX', such as
SerializedItem. They were used to persist catalog objects to the stash
while allowing us to evolve the on disk state independently of the in
memory state.
    
Now that we use protobuf to persist objects to the stash, the
SerializedX structs are no longer necessary. This commit removes these
structs from the catalog. Some of the structs are still necessary to
use as arguments for catalog::storage functions, so those were moved
to the catalog::storage module.

### Motivation
This PR refactors existing code.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - There are no user-facing behavior changes.

### CODE REVIEW

**Review by def- — CHANGES_REQUESTED** [review]
The coverage run ran into a panic in unit test, not sure if related to this change:
```
thread 'test_statement_logging_immediate' panicked at 'assertion failed: `(left == right)`
  left: `"FETCH FORWARD ALL FROM c"`,
 right: `"DECLARE c CURSOR FOR SELECT * FROM t"`', src/environmentd/tests/server.rs:302:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panicking.rs:67:14
   2: core::panicking::assert_failed_inner
   3: core::panicking::assert_failed::<alloc::string::String, alloc::string::String>
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panicking.rs:229:5
   4: server::test_statement_logging_immediate::test_impl
             at ./tests/server.rs:302:9
   5: server::test_statement_logging_immediate
             at ./tests/server.rs:228:1
   6: server::test_statement_logging_immediate::{closure#0}
             at ./tests/server.rs:228:1
   7: <server::test_statement_logging_immediate::{closure#0} as core::ops::function::FnOnce<()>>::call_once
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/ops/function.rs:250:5
   8: core::ops::function::FnOnce::call_once
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/ops/function.rs:250:5
```
Haven't seen it before: https://buildkite.com/materialize/coverage/builds/207

File: src/adapter/src/catalog/storage.rs [hunk 1]
```diff
@@ -2037,15 +2036,110 @@ pub struct ClusterReplica {
...
+}
+
+#[derive(Clone, Debug)]
+pub struct ComputeReplicaLogging {
+    pub log_logging: bool,
+    pub interval: Option<Duration>,
+}
+
+impl From<mz_compute_client::controller::ComputeReplicaLogging> for ComputeReplicaLogging {
+    fn from(logging: mz_compute_client::controller::ComputeReplicaLogging) -> Self {
```
Comment (def-) [inline src/adapter/src/catalog/storage.rs:2130]: This code seems untested, https://buildkite.com/materialize/coverage/builds/216#_

**Review by def- — CHANGES_REQUESTED** [review]

File: src/stash/protos/objects.proto [hunk 2]
```diff
@@ -238,16 +238,6 @@ enum CatalogItemType {
...
-message CatalogItem {
-    message V1 {
-        string create_sql = 1;
-    }
-
-    oneof value {
-        V1 v1 = 1;
-    }
-}
-
```
Comment (ParkMyCar) [inline src/stash/protos/objects.proto:250]: What do you think about keeping the `CatalogItem` message? I feel like one day we might want to persist more than just the `create_sql`, and keeping the message might make that easier? Not sure though

**Review by ParkMyCar — COMMENTED** [review]
All in all this looks good, thank you for cleaning this up!

File: src/adapter/src/catalog/storage.rs (hunk 1 above)
Comment (jkosh44) [reply src/adapter/src/catalog/storage.rs]: Thanks, that was dead code. I've removed it.

**Comment by jkosh44** [comment]
> The coverage run ran into a panic in unit test, not sure if related to this change

@def-, I saw that you opened a separate issue for this. Does that mean that it's confirmed not related?

File: src/stash/protos/objects.proto (hunk 2 above)
Comment (jkosh44) [reply src/stash/protos/objects.proto]: Updated the PR to revert the stash changes.

**Review by def- — APPROVED** [review]
Correct, the same unit test panic also happened outside of this PR, so not related. Thanks for removing the dead code.

**Review by antiguru — APPROVED** [review]
I'm glad to see this! Approving because I think it's a great change.

**Pull request merged**